### ISSUE TITLE
Sigma escape rules differ from glob escape rules enough to cause issues...

### ISSUE THREAD

**Issue by newodahs** [newodahs/go-sigma-rule-engine#5]
The Sigma spec has a certain set of rules for escaping characters (https://github.com/SigmaHQ/sigma/wiki/Specification#escaping).

These rules unfortunately differ from the new glob library in use (gobwas/glob), which follows a more standard method of escaping for its use case (generally: https://tldp.org/LDP/GNU-Linux-Tools-Summary/html/x11655.htm).

The result is that certain sigma rules that have singular backslashes and are in a glob or `contains` modifier (which is a implicit glob on either end) pattern end up having those backslashes misinterpreted (removed), increasing the likelyhood of false positives (especially in the `contains` case).

As an example a`contains` rule string of '\sam' (ref: https://github.com/SigmaHQ/sigma/blob/master/rules/windows/builtin/security/win_transferring_files_with_credential_data_via_network_shares.yml) will do the following:
 1. `contains` detected as a modifier changes '\sam' => '*\sam*'
 1. glob.Compile('*\sam*') results in a compiled glob that looks like: '*sam*'
     * This is because '\s' in the glob library is saying to escape 's'
 1. a selector coming through with something like 'abc sam cba' will match, which is a false positive as the rule really wants to match on '\sam' and not just 'sam' 

Basically, during parsing we need to convert anything with a glob from Sigma's escaping rules to ones that work with the standard glob format.

**Comment by newodahs** [comment]
Fixed with a0cb3372d95bdfd2552d6be4137873632244c424.
